contractcourt/htlc_timeout_resolver.go

```diff
@@ -362,14 +441,87 @@ func (h *htlcTimeoutResolver) sweepSecondLevelTransaction(
 		reports []*channeldb.ResolverReport
 	)
 
+	switch {
+
+	// If the sweeper is handling the second level transaction, wait for
+	// the CSV lock to expire, before sweeping the output on the
+	// second-level.
+	case h.htlcResolution.SignDetails != nil:
+		h.reportLock.Lock()
+		h.currentReport.Stage = 2
+		h.reportLock.Unlock()
+
+		waitHeight := uint32(commitSpend.SpendingHeight) +
+			h.htlcResolution.CsvDelay - 1
+
+		log.Infof("%T(%x): waiting for CSV lock to expire at height %v",
+			h, h.htlc.RHash[:], waitHeight)
+
+		err := waitForHeight(waitHeight, h.Notifier, h.quit)
```

Comment:
nit: much of the logic below looks duplicated from before, not sure if there is a way to reuse

Reply:
The `FindInputIndex` is removed in later commit (will squash with this one), most of the remainder is custom I think.

Reply:
yeah was just looking at everything from `watiForHeight` to `SpendInput`, it is very similar but i think it's fine here